File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy = 0
        sell = 1
        curr_diff = 0
        max_diff = 0
        length = len(prices)
        
        while sell < length:
            curr_diff = prices[sell] - prices[buy]
            if prices[sell] < prices[buy]:
                buy = sell
            else:
                max_diff = max(curr_diff, max_diff)
            sell +=1
        return max_diff

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let currDiff = 0;
    let maxDiff = 0;
    const length = prices.length;
    let buy = 0;
    let sell = 1;
    
    while (sell < length){
        //Check if the difference is greater than the current difference
        if (prices[sell] - prices[buy] > currDiff){
            currDiff = prices[sell] - prices[buy];
        }
        
        if (prices[sell] < prices[buy]){
            buy = sell;
        } else {
            maxDiff = Math.max(currDiff, maxDiff);
        }
        
        sell ++;
        
    }
    
    return maxDiff;
}